### PR TITLE
Release 3.1.0

### PR DESCRIPTION
**Pull request by batouxclass**

### CODE REVIEW

**Comment by batouxclass** [comment]
@kevinmprince @wrprice  could you please address this PR ?

**Comment by wrprice** [comment]
@kevinmprince I've reviewed for PROS proprietary content. Can you please review from an SME perspective?

File: src/environmentsPage.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+const formatError = message => `<br/><li>${message}</li>`;
+
+const validateParams = (json) => {
+  let errorMessages = '';
+
+  if (json) {
+    if (!json.environments) {
+      errorMessages += formatError(
+        'The config.json file does not have a value for the "environments" param',
```
Comment (kevinmprince) [inline src/environmentsPage.js:48]: @dfcortes These error messages don't appear to be internationalized. Can you add an issue review this in the next release.

File: src/environmentsPage.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  process.platform === 'darwin'
+    ? '/pros/desktopclient/launchpage-config.json'
+    : process.env.USERPROFILE + '\\launchpage-config.json';
+
+const getSyntaxError = (error) => {
+  const isExplicit = error instanceof SyntaxError;
+  return `[${isExplicit ? 'EXPLICIT' : 'IMPLICIT'}] ${error.name} in config.json: ${error.message}`;
+};
+
+const formatError = message => `<br/><li>${message}</li>`;
```
Comment (kevinmprince) [inline src/environmentsPage.js:40]: I reviewed the use cases and don't see a direct issue here, but we need to sanitize message for safety

**Review by kevinmprince — APPROVED** [review]
There is a minor issue of XSS, but it looks to be mitigated with hard coded message strings.
I'm approving this and requesting this be addressed in the next release.

File: src/environmentsPage.js (hunk 1 above)
Comment (batouxclass) [reply src/environmentsPage.js:48]: Ok @kevinmprince I'm going to do it

**Pull request merged**